tools/kvmexit_example.txt

```diff
@@ -0,0 +1,242 @@
+things up; and for other cases, then use a percpu_hash.
+
+BTW, we originally use a common hash to do this, with a u64(exit_reason)
+key and a struct exit_info {tgid_pid, exit_reason} value. But due to
+the big lock in bpf_hash, each updating is quite performance consuming.
+
+Now imagine here is a pid_tgidA (vcpu A) exits and is going to run on
+pcpuArrayA, the BPF code flow is as follows:
+
+               pid_tgidA keeps running on the same pcpu
+                        //               \\
+                       //                 \\
+                      // Y               N \\
+                     //                     \\
+             a. cache_hit               b. cache_miss
+(cacheA's pid_tgid matches pid_tgidA)        ||
+                  |                          ||
+                  |                          ||
+   "increase percpu exit_ct and return"      ||
+            [*Note*]                         ||
+                             pid_tgidA ever been exited on pcpuArrayA?
+                                           //   \\
+                                          //     \\
+                                         //       \\
+                                        // Y     N \\
+                                       //           \\
+                          b.a load_last_hashA   b.b initialize_hashA_with_zero
+                                          \       /
+                                           \     /
+                                            \   /
+                                      "increase percpu exit_ct"
+                                             ||
+                                             ||
+                           is another pid_tgid been running on pcpuArrayA?
+                                        //          \\
+                                       // Y        N \\
+                                      //              \\
+                       b.*.a save_theLastHit_hashB    do_nothing
+                                           \\       //
+                                            \\     //
+                                             \\   //
+                                       b.* save_to_pcpuArrayA
+
+
+[*Note*] we do not update the table in above "a.", in case the vcpu hit the same
+pcpu again when exits next time, instead we only update until this pcpu is not
+hitted by the same tgidpid(vcpu) again, which is in "b.*.a" and "b.*".
+
+
+#USAGE message:
+
+# ./kvmexit.py -h
+usage: kvmexit.py [-h] [-p PID [-v VCPU | -a] ] [-t TID | -T 'TID1,TID2'] [duration]
+
+Display kvm_exit_reason and its statistics at a timed interval
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -p PID, --pid PID     display process with this PID only, collpase all tids with exit reasons sorted in descending order
+   -v VCPU, --v VCPU    display this VCPU only for this PID
+   -a, --alltids        display all TIDS for this PID
```

Comment:
some alignment issue here.

Reply:
> some alignment issue here.

Emmm, actually '-v' and '-a' is not aligned on purpose, as they are sub-parameters for '-p'.  But now I am intending to make them aligned,, after all, the "examples" and "usage" have explained all. :)